cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -69,38 +75,47 @@ where
 {
     let mut seen_blocks: HashSet<Hash> = HashSet::new();
 
+    let block = connector
+        .latest_block()
+        .compat()
+        .await?
+        .ok_or_else(|| anyhow::anyhow!("Connector returned null latest block"))?;
+    co.yield_(block.clone()).await;
+
+    let blockhash = block
+        .hash
+        .ok_or_else(|| anyhow::anyhow!("Connector returned latest block with null hash"))?;
+    seen_blocks.insert(blockhash);
+
+    // Look back in time until we get a block that predates start_of_swap.
+    let parent_hash = block.parent_hash;
+    let start = start_of_swap.timestamp() as i64;
```

Comment:
These vars are used immediately after they are defined to look back in time (previous patch change the look back to start with parent hash instead of current block hash).